vmdb/app/models/ems_refresh/parsers/foreman.rb

```diff
@@ -0,0 +1,116 @@
+module EmsRefresh
+  module Parsers
+    class Foreman
+      # we referenced a record that does not exist in the database
+      attr_accessor :missing_key
+
+      def self.provisioning_inv_to_hashes(inv)
+        new.provisioning_inv_to_hashes(inv)
+      end
+
+      def self.configuration_inv_to_hashes(inv)
+        new.configuration_inv_to_hashes(inv)
+      end
+
+      def provisioning_inv_to_hashes(inv)
+        result = {}
+        ids = {}
+        result[:customization_scripts] = customization_scripts_inv_to_hash(inv[:media], inv[:ptables])
+        add_ids(ids, result[:customization_scripts])
+        result[:operating_system_flavors] = operating_system_flavors_inv_to_hashes(inv[:operating_systems], ids)
+        result
+      end
+
+      def configuration_inv_to_hashes(inv)
+        result = {}
+        ids = {}
+        result[:configuration_profiles] = configuration_profile_inv_to_hashes(inv[:hostgroups],
+                                                                              inv[:operating_system_flavors],
+                                                                              inv[:customization_scripts])
+        add_ids(ids, result[:configuration_profiles])
+        result[:configured_systems] = configured_system_inv_to_hashes(inv[:hosts], ids, inv[:operating_system_flavors])
+        result[:missing_key] = true if missing_key
```

Comment:
Won't this key mess up the save_inventory stuff?  I never see it removed anywhere.  I think it's fine here, but should be removed before being passed into save_*_inventory.